spring-integration-core/src/main/java/org/springframework/integration/transformer/SimpleFromAvroTransformer.java

```diff
@@ -37,18 +43,94 @@
  */
 public class SimpleFromAvroTransformer extends AbstractTransformer {
 
-	private final Class<? extends SpecificRecord> type;
+	private final Class<? extends SpecificRecord> defaultType;
+
+	private final Map<String, Class<? extends SpecificRecord>> typeMappings = new HashMap<>();
 
 	private final DecoderFactory decoderFactory = new DecoderFactory();
 
-	public SimpleFromAvroTransformer(Class<? extends SpecificRecord> type) {
-		this.type = type;
+	private Expression typeIdExpression =
+			EXPRESSION_PARSER.parseExpression("headers['" + AvroHeaders.TYPE_ID + "']");
+
+	private EvaluationContext evaluationContext;
+
+	/**
+	 * Construct an instance with the supplied default type to create.
+	 * @param defaultType the type.
+	 */
+	public SimpleFromAvroTransformer(Class<? extends SpecificRecord> defaultType) {
+		Assert.notNull(defaultType, "'defaultType' must not be null");
+		this.defaultType = defaultType;
+	}
+
+	/**
+	 * Set type mappings.
+	 * @param typesToMap the types to map.
+	 * @return the transformer.
+	 * @see #typeIdExpression
+	 */
+	public SimpleFromAvroTransformer typeMappings(Map<String, Class<? extends SpecificRecord>> typesToMap) {
```

Comment:
I wonder why you don't honor XML configuration any more...
It really would be great to have them as `set...`, so end-user still simply can use this class for the `<bean>` definition.

On the other hand why do we need this mapping at all? Why a simple expression/function execution against a message is not enough to get an expected type for deserialization? At least that is exactly what we do in the HTTP/WebFlux/RSocket and many other places.